src/Database/Bolt/Connection.hs

```diff
@@ -84,11 +85,16 @@ pullRecords strict pipe keys = fetch pipe >>= cases
         pure (record:rest)
 
 -- |Sends request to database and makes an action
-sendRequest :: Pipe -> Text -> Map Text Value -> IO Response
-sendRequest pipe cypher params =
-  do flush pipe $ RequestRun cypher params
+sendRequest :: Pipe -> Cypher -> IO Response
+sendRequest pipe c =
+  do flush pipe $ RequestRun (cypherQuery c) (cypherParams c)
      status <- fetch pipe
      if isSuccess status
        then pure status
        else do ackFailure pipe
                mkFailure status
+
+data Cypher = Cypher
```

Comment:
We can make the change to update all the `query` functions to use `Cypher` in a future PR. I felt it was too much so this is only use internally now. If you prefer I can remove it completely as well.